### PR TITLE
feat: add ActivationManagement project and enhance XiansAgent functio…

### PR DESCRIPTION
**Pull request by hasith**
…nality

- Introduced the ActivationManagement project to the solution.
- Added TenantAgents property to XiansAgent for tenant-scoped access to other agents.
- Updated activation status check in XiansAgent to utilize TenantAgents for improved cross-agent functionality.
- Enhanced ActivationValidationService to support cancellation tokens in response content reading.
- Added new constants for activation and agent existence endpoints in WorkflowConstants.

### CODE REVIEW

**Comment by hasith** [comment]
🔍 PR Review in Progress

Claude Code is analyzing this pull request. The review will be posted here shortly.

PR Reviewer (1.9.9)

**Review by hasith — COMMENTED** [review]
## PR Review Report

**PR:** feat: add ActivationManagement project and enhance XiansAgent functionality
**Author:** hasith
**Files Changed:** 18 | **+1602** / **-11**
**Verdict:** APPROVE

---

### Summary
This PR adds a new cross-agent activation SDK (`AgentReference`, `TenantAgents`, `ActivationInfo`, `ActivationWorkflowConfig`) that lets an agent inspect and manage another agent's activations in the same tenant, refactors `XiansAgent.GetActivationStatusAsync` to delegate to the new shared path, and ships a complete `ActivationManagement` example project plus a large, thorough unit test suite (`AgentReferenceTests.cs`, 496 lines). The new library code is well documented, has consistent error handling (typed exceptions for 400/404/409, `HttpRequestException` for transient errors so retry policies apply), and correctly escapes all name/id values used in URLs. Two independent automated finder passes and a manual read of the core library changes (`AgentReference.cs`, `TenantAgents.cs`, `ActivationInfo.cs`, `ActivationValidationService.cs`, `XiansAgent.cs`, `WorkflowConstants.cs`) turned up no critical or warning-level issues — the delegation refactor in `XiansAgent.GetActivationStatusAsync` is functionally equivalent to the code it replaces, and the lone cancellation-token change in `ActivationValidationService.cs` is a straightforward, correct improvement.

---

### Critical Issues (Must Fix)
No critical issues found.

---

### Warnings (Should Fix)
No warnings found.

---

### Suggestions (Consider Improving)
No suggestions — the change is clean as submitted.

---

### Review Details

#### Code Quality
The new `AgentReference`/`TenantAgents`/`ActivationInfo` API surface is small, well-documented with XML doc comments and usage examples, and follows existing SDK conventions (mirrors the webhook/activation-validation HTTP client patterns, e.g. `AddTenantHeader`, `EnsureHttpService`). The `XiansAgent.GetActivationStatusAsync` refactor cleanly removes duplicated HTTP/status-check logic in favor of delegating to `Tenant.Agent(Name).GetActivationStatusAsync(...)`, and `ActivationInfo.Bind`/`ActivateAsync`/`DeactivateAsync` provide a nice fluent handle pattern for list/create results.

#### Security
No injection, auth-bypass, or logging-of-secrets issues found. All agent/activation names and ids are passed through `Uri.EscapeDataString` before being placed in URLs, and `AgentReference`'s constructor rejects names containing `:` to avoid workflow-id delimiter collisions. Tenant header propagation is conditioned on `SystemScoped`, consistent with the rest of the codebase.

#### Test Coverage
`Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs` adds 496 lines covering exists/status/list/create/activate/deactivate happy paths, 404/409/400 error mapping, and constructor validation — coverage for the new SDK surface is thorough.

#### Performance
No N+1, unbounded loop, or blocking-I/O concerns in the changed code. HTTP calls use `GetHealthyClientAsync()` consistent with the rest of the SDK, and the example's heartbeat workflow uses `Workflow.DelayAsync` rather than busy-waiting.

---

### Files Reviewed

| File | Lines Changed | Risk | Notes |
|------|---------------|------|-------|
| `Xians.Lib/Agents/Core/AgentReference.cs` | +381/-0 | 🟡 Medium | New cross-agent HTTP client SDK surface |
| `Xians.Lib/Agents/Core/XiansAgent.cs` | +7/-11 | 🟡 Medium | Refactored to delegate to `TenantAgents` |
| `Xians.Lib/Agents/Workflows/ActivationValidationService.cs` | +1/-1 | 🟢 Low | Passes cancellationToken through |
| `Xians.Lib/Agents/Core/Activations/ActivationInfo.cs` | +119/-0 | 🟢 Low | New DTO + fluent activate/deactivate handle |
| `Xians.Lib/Agents/Core/Activations/ActivationWorkflowConfig.cs` | +44/-0 | 🟢 Low | New DTO |
| `Xians.Lib/Agents/Core/TenantAgents.cs` | +32/-0 | 🟢 Low | Thin factory for `AgentReference` |
| `Xians.Lib/Common/WorkflowConstants.cs` | +4/-0 | 🟢 Low | New endpoint constants |
| `Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs` | +496/-0 | 🟢 Low | New unit tests, thorough coverage |
| `Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleActivities.cs` | +177/-0 | 🟢 Low | Example activity code |
| `Xians.Examples/ActivationManagement/Program.cs` | +110/-0 | 🟢 Low | Example entry point |
| `Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleWorkflow.cs` | +63/-0 | 🟢 Low | Example workflow |
| `Xians.Examples/ActivationManagement/ManagerAgent/ManagerAgentSetup.cs` | +51/-0 | 🟢 Low | Example setup |
| `Xians.Examples/ActivationManagement/TargetAgent/TargetHeartbeatWorkflow.cs` | +33/-0 | 🟢 Low | Example workflow |
| `Xians.Examples/ActivationManagement/TargetAgent/TargetAgentSetup.cs` | +30/-0 | 🟢 Low | Example setup |
| `XiansAi.Lib.sln` | +15/-0 | 🟢 Low | New project registration |
| `Xians.Examples/ActivationManagement/ActivationManagement.csproj` | +18/-0 | 🟢 Low | New project file |
| `Xians.Examples/ActivationManagement/Constants.cs` | +14/-0 | 🟢 Low | Example constants |
| `Xians.Examples/ActivationManagement/env.template` | +5/-0 | 🟢 Low | Example env template |

**Comment by hasith** [comment]
🔍 PR Review in Progress

Claude Code is analyzing this pull request. The review will be posted here shortly.

PR Reviewer (1.9.9)

**Review by hasith — COMMENTED** [review]
## PR Review Report

**PR:** feat: add ActivationManagement project and enhance XiansAgent functionality (#112)
**Author:** hasith
**Files Changed:** 28 | **+2662** / **-11**
**Verdict:** REQUEST CHANGES

---

### Existing review threads

- ✅ Appears addressed: 0 open thread(s)
- ⏳ Still open: 3 open thread(s) — no reply (avoid spam)
- 🔇 Duplicates avoided: 2 finding(s) not re-posted (HttpResponseMessage disposal in `AgentReferenceTests.cs:437,463`, generic catch in `ActivationManagement/Program.cs:101` — already flagged by `github-code-quality`)

---

### Summary
This PR adds a new `AgentReference`/`TenantAgents` core abstraction for cross-agent activation checks, refactors `XiansAgent` to use it, and ships two new example projects (`ActivationManagement`, `CrossTenantOperations`). The core library change is sound and well-tested, but the new `CrossTenantOperations` example has a real access-control gap — it forwards a caller-supplied `tenantId` from an unauthenticated webhook straight into privileged Admin API calls (confused-deputy pattern) — and both new example projects create a fresh `HttpClient`/`XiansAdminApiClient` per activity invocation, a socket-exhaustion risk if used as a reference pattern. Since these are example/reference projects intended to demonstrate patterns for consumers of this library, both issues should be fixed before merge so they aren't copied into production agents.

---

### Critical Issues (Must Fix)

- [ ] `Xians.Examples/CrossTenantOperations/RouterAgent/RouterAgentSetup.cs:43` — The webhook handler parses `tenantId` directly out of the unauthenticated, externally-supplied JSON body (`TryParseRouteRequest`) and passes it straight into a workflow that uses the system-wide Admin API bearer token to act on that tenant. Anyone who can call the Router's webhook can direct the privileged Admin API at **any** tenant in the system, not just their own — a classic confused-deputy / broken access control (OWASP A01) flaw.
  ```csharp
  // Current (problematic)
  if (!TryParseRouteRequest(context.Webhook.Payload, out var tenantId, out var payload, out var error))
  {
      context.Response = WebhookResponse.BadRequest(error);
      return;
  }
  // tenantId now flows, unchecked, into a workflow backed by the admin bearer token.

  // Fix
  if (!TryParseRouteRequest(context.Webhook.Payload, out var tenantId, out var payload, out var error)
      || !IsTenantAuthorizedForCaller(context, tenantId))
  {
      context.Response = WebhookResponse.BadRequest(error ?? "tenantId not authorized for this caller.");
      return;
  }
  ```

- [ ] `Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:14` — `RouteAsync(string tenantId, string payload)` accepts an arbitrary tenant id and forwards it to `ResolveSingleActivationAsync` / `EnsureWebhookAsync` / `InvokeWebhookAsync` using the admin bearer token — the same confused-deputy flaw as above, manifesting as tenant enumeration (activation existence/count leakage) and cross-tenant webhook invocation for any tenant id an external caller chooses.
  ```csharp
  // Fix: enforce a tenant authorization check at the activity boundary before any Admin API call
  if (!await _tenantAuthorizer.IsAuthorizedAsync(tenantId))
  {
      result.Message = $"Caller is not authorized for tenant '{tenantId}'.";
      return result;
  }
  ```

- [ ] `Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:30` — A brand-new `XiansAdminApiClient` (which wraps its own `HttpClient`) is constructed and disposed on **every** `RouteAsync` activity invocation — the classic per-request-`HttpClient` socket-exhaustion anti-pattern, on a hot Temporal activity path.
  ```csharp
  // Current (problematic)
  using var admin = XiansAdminApiClient.FromEnvironment();

  // Fix: inject a single shared instance (e.g. via constructor injection backed by
  // IHttpClientFactory, or a static/singleton XiansAdminApiClient held by the activities class)
  ```

- [ ] `Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:222` — `InvokeWebhookAsync` creates a brand-new `HttpClient` on every call instead of reusing the instance-level `_http` field or a shared/pooled client — the same socket/handle-exhaustion risk as above, and this is the actual hot path invoked once per routed message.
  ```csharp
  // Current (problematic)
  using var invokeClient = new HttpClient { Timeout = TimeSpan.FromSeconds(90) };

  // Fix: reuse a single static/shared HttpClient (e.g. via IHttpClientFactory or a
  // lazily-initialized static instance configured with the 90s timeout)
  ```

---

### Warnings (Should Fix)

- [ ] `Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:37` — The client accepts any `serverUrl` without requiring HTTPS (the shipped `env.template` even defaults to `http://localhost:5005`), so the `Bearer {XIANS_ADMIN_TOKEN}` header could be sent in cleartext if `XIANS_SERVER_URL` is ever misconfigured with `http://` in a non-local environment. Validate/enforce `https://` (allowing `http://localhost`/`http://127.0.0.1` for local dev only), throwing otherwise.
- [ ] `Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:242` — `EnsureSuccessAsync` doesn't accept a `CancellationToken`, so `ReadAsStringAsync()` inside it (and the three call sites at lines 70, 126, 170) can't honor cancellation even though every other method in this class threads a token through. Add a `CancellationToken cancellationToken = default` parameter to `EnsureSuccessAsync`, pass it to `ReadAsStringAsync(cancellationToken)`, and update the call sites.
- [ ] `Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleActivities.cs:99` — `GetActivationStatusAsync` (line 91) and `ActivationExistsAsync` (line 99) are called back-to-back for the same activation; `ActivationExistsAsync` re-issues an identical HTTP GET the code just made a line earlier, doubling network round-trips for no new information (the same redundant pair repeats around lines 146/148 in `DeactivateDemoActivationAsync`). Derive the bool from the already-fetched `status` instead: `result.ActivationExistsAfterActivate = status == ActivationCheckStatus.Active;`
- [ ] `Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs:373` — `AgentReference`'s constructor (`AgentReference.cs:43`), `CreateActivationAsync` (`AgentReference.cs:178`), `ActivateAsync` (`AgentReference.cs:229`) and `DeactivateAsync` (`AgentReference.cs:274`) all validate their name/id parameters with `string.IsNullOrWhiteSpace`, but only the "colon in name" and one "empty activation name" case are exercised in `AgentReferenceTests.cs` — there's no test for `Agent(null)`/`Agent("")`, or null/empty `name`/`activationId` in `CreateActivationAsync`/`ActivateAsync`/`DeactivateAsync`. Add parameterized (`[Theory]`/`[InlineData(null)]`/`[InlineData("")]`/`[InlineData("  ")]`) tests asserting `ArgumentException` for each.

---

### Suggestions (Consider Improving)

- [ ] `Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:118` — `RouteResult.ProcessorResponseBody` (up to 500 chars of the Processor Agent's response, which — until the access-control issue above is fixed — may belong to a different tenant than the Router's own) is persisted into durable Temporal workflow history, giving Router-tenant operators visibility into another tenant's response data. Consider dropping or further redacting this field.
- [ ] `Xians.Examples/CrossTenantOperations/ProcessorAgent/ProcessorAgentSetup.cs:50` — Generic `catch (Exception ex)` around the webhook handler logs the full exception to console; prefer catching narrower expected exception types and logging a sanitized message.
- [ ] `Xians.Examples/ActivationManagement/Program.cs:45` — `targetAgent.UploadWorkflowDefinitionsAsync()` and `managerAgent.UploadWorkflowDefinitionsAsync()` are independent remote calls awaited sequentially; consider `Task.WhenAll(...)`.
- [ ] `Xians.Examples/CrossTenantOperations/Program.cs:49` — Same pattern: `processorAgent.UploadWorkflowDefinitionsAsync()` and `routerAgent.UploadWorkflowDefinitionsAsync()` could run concurrently via `Task.WhenAll(...)`.

---

### Review Details

#### Code Quality
Overall clean, well-documented additions. `AgentReference`/`TenantAgents` provide a clear fluent API (`agent.Tenant.Agent("Other").ExistsAsync()`) that appropriately encapsulates HTTP details, and the `XiansAgent.GetActivationStatusAsync` refactor to delegate to `TenantAgents` removes duplication. The main gap is `EnsureSuccessAsync` not threading its `CancellationToken` (see Warnings). Two issues already flagged by the `github-code-quality` bot (generic catch in `ActivationManagement/Program.cs:101`, undisposed `HttpResponseMessage` in test helpers at `AgentReferenceTests.cs:437,463`) are left to that existing thread rather than duplicated here.

#### Security
The `AgentReference`/`TenantAgents` abstraction itself only ever operates within the current tenant's execution context (tenant header derived from trusted framework context, not user input) and is not the vulnerable surface. The real risk is in the new `CrossTenantOperations` example: a system-wide Admin API bearer token is combined with a tenant id taken directly from an unauthenticated external webhook caller, enabling confused-deputy cross-tenant access (see Critical Issues). A secondary, lower-severity issue is that the Admin API client doesn't require HTTPS, risking cleartext transmission of the bearer token if misconfigured. No hardcoded secrets, injection, or unsafe deserialization were found; all user-supplied URL segments are properly escaped via `Uri.EscapeDataString`.

#### Test Coverage
`AgentReferenceTests.cs` (496 lines, new) gives strong coverage of the new `AgentReference`/`TenantAgents` surface — happy paths, 400/404/409/500 handling, system-scoped vs. non-system-scoped tenant header behavior — roughly 95% of new public methods. The refactored `XiansAgent.GetActivationStatusAsync` delegation is covered by existing `WebhookCollectionTests.cs`. The gap is missing null/empty/whitespace parameter-validation tests across several methods that do validate those inputs in production code (see Warnings) — a straightforward, low-cost addition.

#### Performance
Two genuine socket-exhaustion risks: a new `XiansAdminApiClient` (and its `HttpClient`) is constructed per Temporal activity invocation in `RouteToProcessorActivities.RouteAsync`, and `XiansAdminApiClient.InvokeWebhookAsync` additionally allocates its own throwaway `HttpClient` per call rather than reusing the instance's own `_http` field (see Critical Issues). Beyond that, one redundant HTTP round-trip in the `ActivationManagement` heartbeat/lifecycle demo (Warnings) and two minor sequential-await-that-could-be-parallel spots (Suggestions). The core library changes (`AgentReference`/`TenantAgents`, the `ActivationValidationService` cancellation-token addition, the heartbeat delay loop) introduce no new N+1, blocking, or unbounded-retry issues.

---

### Files Reviewed

| File | Lines Changed | Risk | Notes |
|------|---------------|------|-------|
| `Xians.Examples/CrossTenantOperations/RouterAgent/RouterAgentSetup.cs` | +149 | 🔴 High | Untrusted tenantId enters here |
| `Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs` | +160 | 🔴 High | Confused-deputy + per-call HttpClient |
| `Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs` | +378 | 🔴 High | Admin bearer token, per-call HttpClient, no HTTPS enforcement |
| `Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleActivities.cs` | +177 | 🟡 Medium | Redundant HTTP call |
| `Xians.Lib/Agents/Core/AgentReference.cs` | +381 | 🟡 Medium | Core library — well tested |
| `Xians.Lib/Agents/Core/TenantAgents.cs` | +32 | 🟢 Low | Thin delegation |
| `Xians.Lib/Agents/Core/XiansAgent.cs` | +19/-11 (net) | 🟢 Low | Delegates to TenantAgents, covered by existing tests |
| `Xians.Lib/Agents/Workflows/ActivationValidationService.cs` | +2/-1 | 🟢 Low | Cancellation token correctly threaded |
| `Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs` | +496 | 🟢 Low | Strong coverage; missing validation-edge tests |
| `Xians.Examples/CrossTenantOperations/ProcessorAgent/ProcessorAgentSetup.cs` | +62 | 🟡 Medium | Generic catch clause |
| `Xians.Examples/ActivationManagement/Program.cs` | +110 | 🟢 Low | Sequential awaits (suggestion) |
| `Xians.Examples/CrossTenantOperations/Program.cs` | +86 | 🟢 Low | Sequential awaits (suggestion) |
| …and 16 more files | +598/-0 | 🟢 Low | Project scaffolding (`.csproj`, `.sln`), constants, README, env templates, workflow/setup boilerplate |

File: Xians.Examples/CrossTenantOperations/RouterAgent/RouterAgentSetup.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        var integrator = agent.Workflows.DefineIntegrator();
+        integrator.OnWebhook(async (context) =>
+        {
+            try
+            {
+                Console.WriteLine(
+                    $"[Router Agent] Webhook '{context.Webhook.Name}' received "
+                    + $"(payload length={context.Webhook.Payload?.Length ?? 0}).");
+
+                if (!TryParseRouteRequest(context.Webhook.Payload, out var tenantId, out var payload, out var error))
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/RouterAgent/RouterAgentSetup.cs:43]: **CRITICAL:** `tenantId` is parsed straight out of the unauthenticated, externally-supplied webhook JSON body and passed unchecked into a workflow that uses the system-wide `XIANS_ADMIN_TOKEN` to act on that tenant — a confused-deputy / broken access control (OWASP A01) flaw: anyone who can call the Router's webhook can direct the privileged Admin API at any tenant in the system, not just their own.

**Fix:** Validate the caller-supplied `tenantId` against an allowlist of tenants the Router webhook/caller is authorized for (e.g. derive it from the webhook's own scoped credential/claims rather than trusting the request body), and reject/audit requests for tenants outside that set before starting `RouteToProcessorWorkflow`.

File: Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+namespace Xians.Examples.RouterProcessor.RouterAgent;
+
+/// <summary>
+/// Activities that call the Xians Admin API (HTTP I/O — must not run in deterministic workflow code).
+/// Sequence: resolve single Processor activation → ensure named webhook → invoke webhook with payload.
+/// </summary>
+public class RouteToProcessorActivities
+{
+    [Activity]
+    public async Task<RouteResult> RouteAsync(string tenantId, string payload)
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:14]: **CRITICAL:** `RouteAsync(string tenantId, string payload)` accepts an arbitrary tenant id and forwards it to `ResolveSingleActivationAsync`/`EnsureWebhookAsync`/`InvokeWebhookAsync` using the admin bearer token — the same confused-deputy flaw as `RouterAgentSetup.cs:43`, enabling tenant enumeration (activation existence/count leakage) and cross-tenant webhook invocation for any tenant id an external caller chooses.

**Fix:** Enforce a tenant authorization check at the activity boundary (e.g. verify `tenantId` against a per-caller allowlist or a signed claim from the inbound webhook) before making any Admin API call, and fail closed when the caller has no explicit grant for that tenant.

File: Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            WebhookName = Constants.ProcessorWebhookName
+        };
+
+        if (string.IsNullOrWhiteSpace(tenantId))
+        {
+            result.Message = "tenantId is required.";
+            return result;
+        }
+
+        using var admin = XiansAdminApiClient.FromEnvironment();
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:30]: **CRITICAL:** A brand-new `XiansAdminApiClient` (which wraps its own `HttpClient`) is constructed and disposed on every single `RouteAsync` activity invocation — the classic per-request-`HttpClient` socket-exhaustion anti-pattern — and this activity runs once per routed message.

**Fix:** Inject a single shared `HttpClient`/`XiansAdminApiClient` (e.g. via `IHttpClientFactory` or a static/singleton instance held by the activities class) instead of creating one per activity execution.

File: Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+    {
+        if (string.IsNullOrWhiteSpace(webhookUrl))
+            throw new ArgumentException("Webhook URL is required.", nameof(webhookUrl));
+
+        var absolute = webhookUrl.StartsWith("http", StringComparison.OrdinalIgnoreCase)
+            ? webhookUrl
+            : _serverUrl + (webhookUrl.StartsWith('/') ? webhookUrl : "/" + webhookUrl);
+
+        // Use a plain HttpClient (no Bearer header) — builtin webhooks authenticate via apikey in the URL.
+        using var invokeClient = new HttpClient { Timeout = TimeSpan.FromSeconds(90) };
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:222]: **CRITICAL:** `InvokeWebhookAsync` creates a brand-new `HttpClient` on every call instead of reusing the instance-level `_http` field or a shared/pooled client, causing socket/handle exhaustion under real load — this is the hot path invoked once per routed message via `RouteToProcessorActivities.RouteAsync`.

**Fix:** Reuse a single static/shared `HttpClient` (e.g. `HttpClientFactory.CreateClient` or a lazily-created static instance with the 90s timeout) instead of `new HttpClient(...)` per invocation.

File: Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+            throw new ArgumentException("Server URL is required.", nameof(serverUrl));
+        if (string.IsNullOrWhiteSpace(adminToken))
+            throw new ArgumentException("Admin token is required.", nameof(adminToken));
+
+        _serverUrl = serverUrl.TrimEnd('/');
+        var token = adminToken.Trim();
+        if (token.StartsWith("Bearer ", StringComparison.OrdinalIgnoreCase))
+            token = token["Bearer ".Length..].Trim();
+
+        _http = new HttpClient { BaseAddress = new Uri(_serverUrl + "/") };
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:37]: **WARNING:** The client accepts any `serverUrl` without requiring HTTPS (the shipped `env.template` even defaults to `http://localhost:5005`), so the `Bearer {XIANS_ADMIN_TOKEN}` header could be sent in cleartext if `XIANS_SERVER_URL` is ever misconfigured with `http://` in a non-local environment.

**Fix:** Validate/enforce that `serverUrl` starts with `https://` (allow `http://localhost` / `http://127.0.0.1` only for local dev), throwing if an admin token would otherwise be sent over plaintext HTTP.

File: Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+    // -------------------------------------------------------------------------
+    // Helpers
+    // -------------------------------------------------------------------------
+
+    private static async Task EnsureSuccessAsync(HttpResponseMessage response, string operation)
+    {
+        if (response.IsSuccessStatusCode)
+            return;
+
+        var body = await response.Content.ReadAsStringAsync();
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/Admin/XiansAdminApiClient.cs:242]: **WARNING:** `EnsureSuccessAsync` doesn't accept a `CancellationToken`, so `ReadAsStringAsync()` inside it can't honor cancellation even though every other method in this class threads a token through (call sites at lines 70, 126, 170 all call it without a token).

**Fix:** Add a `CancellationToken cancellationToken = default` parameter to `EnsureSuccessAsync`, pass it to `ReadAsStringAsync(cancellationToken)`, and update the three call sites to pass their token through.


```suggestion
    private static async Task EnsureSuccessAsync(HttpResponseMessage response, string operation, CancellationToken cancellationToken = default)
    {
        if (response.IsSuccessStatusCode)
            return;

        var body = await response.Content.ReadAsStringAsync(cancellationToken);
        throw new HttpRequestException(
            $"Admin API {operation} failed. Status: {(int)response.StatusCode} {response.StatusCode}. Body: {body}");
    }
```

File: Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleActivities.cs [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        // 4) Status check (Active / NotFound / Deactivated).
+        var status = await target.GetActivationStatusAsync(Constants.DemoActivationName);
+        result.StatusAfterActivate = status.ToString();
+        logger.LogInformation(
+            "[4/7] GetActivationStatusAsync('{Name}') => {Status}",
+            Constants.DemoActivationName,
+            status);
+
+        // 5) Convenience bool.
+        result.ActivationExistsAfterActivate = await target.ActivationExistsAsync(Constants.DemoActivationName);
```
Comment (hasith) [inline Xians.Examples/ActivationManagement/ManagerAgent/ActivationLifecycleActivities.cs:99]: **WARNING:** `GetActivationStatusAsync` (line 91) and `ActivationExistsAsync` (line 99) are called back-to-back for the same activation; `ActivationExistsAsync` re-issues an identical HTTP GET the code just made a line earlier, doubling network round-trips for no new information (the same redundant pair repeats around lines 146/148 in `DeactivateDemoActivationAsync`).

**Fix:** Derive `ActivationExistsAfterActivate` from the already-fetched `status` instead of making a second HTTP call.


```suggestion
        result.ActivationExistsAfterActivate = status == ActivationCheckStatus.Active;
```

File: Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+        await owner.Tenant.Agent(TARGET_AGENT_NAME).ExistsAsync();
+
+        Assert.NotNull(captured);
+        Assert.False(captured!.Headers.Contains(WorkflowConstants.Headers.TenantId));
+    }
+
+    // ---- Construction ----
+
+    [Fact]
+    public void Agent_WithColonInName_Throws()
```
Comment (hasith) [inline Xians.Lib.Tests/UnitTests/Agents/AgentReferenceTests.cs:373]: **WARNING:** `AgentReference`'s constructor (`AgentReference.cs:43`), `CreateActivationAsync` (`AgentReference.cs:178`), `ActivateAsync` (`AgentReference.cs:229`) and `DeactivateAsync` (`AgentReference.cs:274`) all validate their name/id parameters with `string.IsNullOrWhiteSpace`, but only the "colon in name" and one "empty activation name" case are exercised in this file — there's no test for `Agent(null)`/`Agent("")`, or null/empty `name`/`activationId` in `CreateActivationAsync`/`ActivateAsync`/`DeactivateAsync`.

**Fix:** Add parameterized tests (`[Theory]`/`[InlineData(null)]`/`[InlineData("")]`/`[InlineData("  ")]`) asserting `ArgumentException` for each of these validated parameters.

File: Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+            result.Message =
+                $"Webhook '{Constants.ProcessorWebhookName}' has no webhookUrl; cannot invoke.";
+            return result;
+        }
+
+        // 3) Invoke the Processor webhook with the payload.
+        var (statusCode, body) = await admin.InvokeWebhookAsync(webhook.WebhookUrl, payload);
+        result.Invoked = true;
+        result.ProcessorStatusCode = statusCode;
+        result.ProcessorResponseBody = Truncate(body, 500);
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/RouterAgent/RouteToProcessorActivities.cs:118]: **SUGGESTION:** `RouteResult.ProcessorResponseBody` (up to 500 chars of the Processor Agent's response, which — until the access-control issue in `RouterAgentSetup.cs:43` is fixed — may belong to a different tenant than the Router's own) is persisted into durable Temporal workflow history, giving Router-tenant operators visibility into another tenant's response data.

**Fix:** Drop `ProcessorResponseBody` from the persisted `RouteResult` (or redact/limit it further) once tenant authorization is enforced, since workflow history is durable and may be broadly viewable within the Router's own tenant.

File: Xians.Examples/CrossTenantOperations/ProcessorAgent/ProcessorAgentSetup.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+                {
+                    processed = true,
+                    agent = Constants.ProcessorAgentName,
+                    webhookName,
+                    tenantId,
+                    payloadLength,
+                    message = "Payload processed by Processor Agent."
+                });
+            }
+            catch (Exception ex)
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/ProcessorAgent/ProcessorAgentSetup.cs:50]: **SUGGESTION:** Generic `catch (Exception ex)` around the Integrator webhook handler logs the full exception to console, which could expose unexpected internal details.

**Fix:** Catch specific expected exception types and log only sanitized messages; reserve a narrowly-scoped generic catch solely for returning a safe 500 response.

File: Xians.Examples/ActivationManagement/Program.cs [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+});
+
+// Register both agents (target first so its definitions exist before the manager activates it).
+var targetAgent = TargetAgentSetup.Setup(xiansPlatform);
+var managerAgent = ManagerAgentSetup.Setup(xiansPlatform);
+
+// Upload definitions so the server knows about both agents before we create activations.
+Console.WriteLine("Uploading workflow definitions...");
+await targetAgent.UploadWorkflowDefinitionsAsync();
+await managerAgent.UploadWorkflowDefinitionsAsync();
```
Comment (hasith) [inline Xians.Examples/ActivationManagement/Program.cs:45]: **SUGGESTION:** `targetAgent.UploadWorkflowDefinitionsAsync()` and `managerAgent.UploadWorkflowDefinitionsAsync()` are independent remote calls but are awaited sequentially.

**Fix:** Run them concurrently with `Task.WhenAll`.


```suggestion
await Task.WhenAll(
    targetAgent.UploadWorkflowDefinitionsAsync(),
    managerAgent.UploadWorkflowDefinitionsAsync());
```

File: Xians.Examples/CrossTenantOperations/Program.cs [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    ApiKey = xiansApiKey,
+    ServerLogLevel = LogLevel.Information,
+});
+
+var processorAgent = ProcessorAgentSetup.Setup(xiansPlatform);
+var routerAgent = RouterAgentSetup.Setup(xiansPlatform);
+
+Console.WriteLine("Uploading workflow definitions...");
+await processorAgent.UploadWorkflowDefinitionsAsync();
+await routerAgent.UploadWorkflowDefinitionsAsync();
```
Comment (hasith) [inline Xians.Examples/CrossTenantOperations/Program.cs:49]: **SUGGESTION:** `processorAgent.UploadWorkflowDefinitionsAsync()` and `routerAgent.UploadWorkflowDefinitionsAsync()` are independent remote calls but are awaited sequentially.

**Fix:** Run them concurrently with `Task.WhenAll`.


```suggestion
await Task.WhenAll(
    processorAgent.UploadWorkflowDefinitionsAsync(),
    routerAgent.UploadWorkflowDefinitionsAsync());
```
